core_contracts/governance/governance.py

```diff
@@ -90,34 +173,53 @@ def cancelVote(self, name: str) -> None:
         proposal.status.set(ProposalStatus.STATUS[ProposalStatus.CANCELLED])
 
     @external
-    @only_owner
-    def defineVote(self, name: str, description: str, quorum: int, vote_start: int, duration: int, snapshot: int, actions: str,
-                   majority: int = MAJORITY) -> None:
+    def defineVote(self, name: str, description: str, vote_start: int, duration: int, 
+                   snapshot: int, actions: str) -> None:
         """
-        Names a new vote, defines quorum, and actions.
+        Define a new vote and which actions are to be executed if it is successful.
+
+        Parameters:
+        name         -   Name of the vote.
+        description  -   Description of the vote.
+        vote_start   -   Day to start the vote.
+        duration     -   Number of days the vote will be active.
+        snapshot     -   Which day to use for the baln stake snapshot.
+        actions      -   Json string on the form:  {'<action_1>': {<kwargs for action_1>},
+                         {'<action_2>': {kwargs_for_action_2}}, ...}
         """
         if len(description) > 500:
             revert(f'Description must be less than or equal to 500 characters.')
-        if not 0 < quorum < 100:
-            revert(f'Quorum must be greater than 0 and less than 100.')
-        if vote_start <= self.getDay():
+        if vote_start < self.getDay():
             revert(f'Vote cannot start before the current time.')
-        if snapshot >= vote_start:
-            revert(f'Snapshot reference index must be less than vote start.')
+        if not self.getDay() <= snapshot <= vote_start:
+            revert(f'The reference snapshot must be in the range: [current_day ({self.getDay()}), '
+                   f'start_day ({vote_start})].')
         min_duration = self._minimum_vote_duration.get()
         if duration < min_duration:
             revert(f'Votes must have a minimum duration of {min_duration} days.')
         vote_index = ProposalDB.proposal_id(name, self.db)
         if vote_index > 0:
             revert(f'Poll name {name} has already been used.')
+        
+        # Test baln staking criteria.
+        baln = self.create_interface_score(self.addresses['baln'], BalancedInterface)
+        baln_total = baln.totalSupply()
+        user_staked = baln.stakedBalanceOf(self.msg.sender)
+        baln_criteria = self._baln_vote_definition_criteria.get()
+        if (POINTS * user_staked) // baln_total < baln_criteria:
+            revert(f'User needs atleast {baln_criteria / 100}% of total baln supply staked to define a vote.')
+
+        # Transfer bnUSD fee to daofund.
+        bnusd = self.create_interface_score(self.addresses['bnUSD'], AssetInterface)
+        bnusd.transfer(self.addresses['daofund'], self._bnusd_vote_definition_fee.get())
```

Comment:
After reviewing this one last time I realized that this call to the bnusd contract will make the governance contract the sender, and not the user. 

I see two different solutions here:

1. Use tokenfallback like you do in the dex contract when a user sends irc2 tokens to it. Pass method and arguments via the data field in the bnUSD contract. Then call defineVote in the tokenFallback method.
2. Add an additional transfer method in the bnUSD contract which uses tx.origin instead of msg.sender.

Something like: 

	@external
	def transfer_origin(self, _to: Address, _value: int, _data: bytes = None):
		if _data is None:
			_data = b'None'
		self._transfer(self.tx.origin, _to, _value, _data)

Option 2 would be easier to implement. Not seen this done before so not sure it has any drawback.
Thoughts?

Reply:
Yeah option 2 looks good and easier. we can implement that.

Reply:
Good catch. Yeah, I'd agree that is the better design for this case. A govTransfer method may be used for some of the other governance functions we will be adding as well. Please add that to the PR.

Reply:
I'll do that. Thanks.